app/views/public_recipes/index.html.erb

```diff
@@ -0,0 +1,25 @@
+<section class="hero">
+  <h1>Public recipes list</h1>
+  <% if @recipes.empty? %>
+    <p>No public recipes for the moment!</p>
+  <% else %>
+  <ul>
+    <% @recipes.includes(:user, :recipe_foods).each do |recipe| %>
+      <li class="public-recipe-link">
+        <%= link_to recipe_path(recipe.id) do %>
+          <div class="public-recipe">
+            <div class="recipe-owner">
+              <p><%= recipe.name %></p>
+              <p>by <%= recipe.user.name %></p>
+            </div>
+            <div class="recipe-details">
+              <p>Total food items: <%= recipe.recipe_foods.length %> </p>
+              <p>Total price: $<%= Recipe.calculate_total_amount(recipe) %></p>
+            </div>
+          </div>
+        <% end %>
+      </li>
+    <% end %>
+  </ul>
+  <% end %>
+</section>
```

Comment:
- [ ] Clicking on a public recipe should lead to the recipe show page even if the User clicking on the recipe did not create it. Currently, this brings about an error, kindly fix this.

Video Demo
[cinnamon-2023-11-26T000332+0200.webm](https://github.com/firepicaso/ror-recipe-app/assets/81932478/9700827c-32e1-4b34-9910-a6d1c689f03d)